pkg/storage/fs/ocis/upload.go

```diff
@@ -41,6 +47,8 @@ import (
 
 var defaultFilePerm = os.FileMode(0664)
 
+// TODO Upload (and InitiateUpload) needs a way to receive the expected checksum.
+// Maybe in metadata as 'checksum' => 'sha1 aeosvp45w5xaeoe' = lowercase, space separated?
```

Comment:
worth tracking as technical debt? I fear this will get lost in the comments

Reply:
would be nice to end up with a list of new issues out of this PR that adds technical debt. I personally don't go through my TODO's so often, although there is great value in doing so, for the sake of the project. I feel there are lots of TODO comments unadressed.

Reply:
tracked in https://github.com/cs3org/cs3apis/issues/103